test/src/module/common/walFilterTest.c

```diff
@@ -0,0 +1,1848 @@
+/***********************************************************************************************************************************
+Test wal filter
+***********************************************************************************************************************************/
+
+#include "command/archive/get/file.h"
+#include "common/harnessConfig.h"
+#include "common/harnessFork.h"
+#include "common/harnessInfo.h"
+#include "common/harnessPostgres.h"
+#include "common/harnessStorage.h"
+#include "common/harnessWal.h"
+#include "common/io/bufferRead.h"
+#include "common/io/io.h"
+#include "common/partialRestore.h"
+#include "common/type/json.h"
+#include "common/walFilter/versions/recordProcessGPDB6.h"
+#include "info/infoArchive.h"
+#include "postgres/interface/crc32.h"
+#include "storage/posix/storage.h"
+
+#define DEFAULT_GDPB_XLOG_PAGE_SIZE 32768
+#define DEFAULT_GDPB_PAGE_SIZE 32768
+
+typedef enum WalFlags
+{
+    NO_SWITCH_WAL = 1 << 0,
+} WalFlags;
+
+typedef struct XRecordInfo
+{
+    uint8_t rmid;
+    uint8_t info;
+    uint32_t body_size;
+    void *body;
+} XRecordInfo;
+
+typedef struct buildWalParam
+{
+    VAR_PARAM_HEADER;
+    PgPageSize pageSize;
+} buildWalParam;
+
+static Buffer *
+testFilter(IoFilter *filter, Buffer *wal, size_t inputSize, size_t outputSize)
+{
+    Buffer *filtred = bufNew(1024 * 1024);
+    Buffer *output = bufNew(outputSize);
+    ioBufferSizeSet(inputSize);
+
+    IoRead *read = ioBufferReadNew(wal);
+    ioFilterGroupAdd(ioReadFilterGroup(read), filter);
+    ioReadOpen(read);
+
+    while (!ioReadEof(read))
+    {
+        ioRead(read, output);
+        bufCat(filtred, output);
+        bufUsedZero(output);
+    }
+
+    ioReadClose(read);
+    bufFree(output);
+    ioFilterFree(filter);
+
+    return filtred;
+}
+
+#define buildWalP(wal, records, count, flags, ...) \
+    buildWal(wal, records, count, flags, (buildWalParam){VAR_PARAM_INIT, __VA_ARGS__})
+
+static __attribute__((unused)) void
+buildWal(Buffer *wal, XRecordInfo *records, size_t count, WalFlags flags, buildWalParam param)
+{
+    if (param.pageSize == 0)
+    {
+        param.pageSize = DEFAULT_GDPB_XLOG_PAGE_SIZE;
+    }
+
+    for (size_t i = 0; i < count; ++i)
+    {
+        XLogRecord *record = hrnGpdbCreateXRecord(records[i].rmid, records[i].info, records[i].body_size, records[i].body);
+        hrnGpdbWalInsertXRecordP(wal, record, NO_FLAGS, .walPageSize = param.pageSize);
+    }
+    if (!(flags & NO_SWITCH_WAL))
+    {
+        XLogRecord *record = hrnGpdbCreateXRecord(0, XLOG_SWITCH, 0, NULL);
+        hrnGpdbWalInsertXRecordP(wal, record, NO_FLAGS, .walPageSize = param.pageSize);
+        size_t to_write = param.pageSize - bufUsed(wal) % param.pageSize;
+        memset(bufRemainsPtr(wal), 0, to_write);
+        bufUsedInc(wal, to_write);
+    }
+}
+
+static void
+fillLastPage(Buffer *wal, PgPageSize pageSize)
+{
+    if (bufUsed(wal) % pageSize == 0)
+    {
```

Comment:
The curly braces braces usage in case of single operator is not consistent along this file. Below braces are not used:
```
    if (expect_not_skip)
        TEST_RESULT_BOOL(memcmp(&node_expect, node_result, sizeof(RelFileNode)), 0, "RelFileNode is different from expected");
    else
        TEST_RESULT_PTR(node_result, NULL, "node_result is not empty");
    memFree(record);
```

Please make the usage consistent.